common/src/main/java/com/lightstep/tracer/shared/Span.java

```diff
@@ -84,11 +76,8 @@ public Span setTag(String key, Number value) {
             return this;
         }
         synchronized (mutex) {
-            if (isJoinKey(key)) {
-                record.addToJoin_ids(new TraceJoinId(key, value.toString()));
-            } else {
-                record.addToAttributes(new KeyValue(key, value.toString()));
-            }
+            // TODO convert to number value? lose precision?
```

Comment:
Alright, will do that then. I was originally more concerned about losing precision if say the number was a BigDecimal, using string will preserve the precision while using double won't. On second thought I can just convert Long and Double classes for now, and use string for all other number classes not explicitly defined in protobuf.